pkg/kubevirt-api-lifecycle-automation/resources/operator/operator.go

```diff
@@ -0,0 +1,185 @@
+			args.NamespacedArgs.TargetNamespace,
+			args.NamespacedArgs.RestartRequired,
+			args.NamespacedArgs.LabelSelector,
+			args.NamespacedArgs.Verbosity,
+			args.Image,
+			args.NamespacedArgs.PullPolicy),
+	}
+}
+
+func createJobEnvVar(machineTypeGlob, targetNamespace, restartRequired, labelSelector, verbosity string) []corev1.EnvVar {
+	envVar := []corev1.EnvVar{
+		{
+			Name:  "MACHINE_TYPE_GLOB",
+			Value: machineTypeGlob,
+		},
+		{
+			Name:  "RESTART_REQUIRED",
+			Value: restartRequired,
+		},
+		{
+			Name:  "VERBOSITY",
+			Value: verbosity,
+		},
+	}
+	if labelSelector != "" {
+		envVar = append(envVar, corev1.EnvVar{Name: "LABEL_SELECTOR", Value: labelSelector})
+	}
+	if targetNamespace != "" {
+		envVar = append(envVar, corev1.EnvVar{Name: "NAMESPACE", Value: targetNamespace})
+	}
+	return envVar
+}
+
+func CreateKubevirtApiLifecycleAutomation(namespace, machineTypeGlob, targetNamespace, restartRequired, labelSelector, verbosity, kubevirtApiLifecycleAutomationImage, pullPolicy string) *v1.Job {
+	container := corev1.Container{
+		Name:            "kubevirt-api-lifecycle-automation",
+		Image:           kubevirtApiLifecycleAutomationImage,
+		ImagePullPolicy: corev1.PullPolicy(pullPolicy),
+		Resources: corev1.ResourceRequirements{
+			Requests: corev1.ResourceList{
+				corev1.ResourceCPU:    resource.MustParse("100m"),
+				corev1.ResourceMemory: resource.MustParse("50M"),
+			},
+		},
+		SecurityContext: &corev1.SecurityContext{
+			Privileged:               ptr.To(false),
+			AllowPrivilegeEscalation: ptr.To(false),
+			RunAsNonRoot:             ptr.To(true),
+			SeccompProfile: &corev1.SeccompProfile{
+				Type: corev1.SeccompProfileTypeRuntimeDefault,
+			},
+			Capabilities: &corev1.Capabilities{
+				Drop: []corev1.Capability{"ALL"},
+			},
+		},
+	}
+	container.Env = createJobEnvVar(machineTypeGlob, targetNamespace, restartRequired, labelSelector, verbosity)
+
+	labels := resources.WithLabels(map[string]string{"name": "kubevirt-api-lifecycle-automation"}, utils2.JobLabels)
+	cj := &v1.Job{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       "Job",
+			APIVersion: "batch/v1",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "kubevirt-api-lifecycle-automation",
+			Namespace: namespace,
+			Labels:    labels,
+		},
+		Spec: v1.JobSpec{
+			Suspend: boolPtr(true),
+			Template: corev1.PodTemplateSpec{
+				Spec: corev1.PodSpec{
+					ServiceAccountName:            "kubevirt-api-lifecycle-automation",
+					HostPID:                       true,
+					HostUsers:                     boolPtr(true),
+					TerminationGracePeriodSeconds: int64Ptr(5),
+					Containers:                    []corev1.Container{container},
+					PriorityClassName:             "system-node-critical",
+					RestartPolicy:                 corev1.RestartPolicyNever,
+				},
+			},
+		},
+		Status: v1.JobStatus{},
+	}
+
+	return cj
+}
+
+func boolPtr(b bool) *bool {
+	return &b
+}
+
+func int64Ptr(i int64) *int64 {
+	return &i
+}
```

Comment:
do we still need those, or we can switch the the `pointer` package?

Reply:
Done

Reply:
@Barakmor1 Did you decide to switch?

Reply:
Yes in the last commit